lib/catalog/determine_task_relevancy.rb

```diff
@@ -26,36 +26,34 @@ def process
 
     private
 
-    def validate_task
-      return true if order_item
-
-      Rails.logger.info("Incoming task does not belong to this app")
-      false
+    def delegate_task
+      if @task.status == "error"
+        if @task.state == "running"
+          Rails.logger.error("Incoming task #{@task.id} had an error while running: #{@task.context}")
+        elsif @task.state == "completed"
+          determine_if_context_is_relevant
+          Rails.logger.error("Incoming task #{@task.id} is completed but errored: #{@task.context}")
+          @order_item.mark_failed
+        end
+      else
+        # Status is either 'warn' or 'ok'
+        determine_if_context_is_relevant
+      end
     end
 
-    def delegate_task
+    def determine_if_context_is_relevant
```

Comment:
Is `process_relevant_context` or `process_if_context_is_relevant` or `delegate_from_context` a better name?

Reply:
Yeah I like `process_relevant_context` 👍